Utils/src/PhotonIDisoProducer.cc

```diff
@@ -254,22 +257,33 @@ PhotonIDisoProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
         // loop over gen particles and find nonprompt and hadronization photons
         int matchedGenPrompt = 0;
         int matchedGenNonPrompt = 0 ;
-        
+	bool photonMatchGenE = false;        
         for( View<reco::GenParticle>::const_iterator iGen = genParticles->begin();
              iGen != genParticles->end();
              ++iGen){
-        
+	  
           // check for non-prompt photons ----------------------
           if( iGen->pdgId() == 22 && ( ( iGen->status() / 10 ) == 2 || iGen->status() == 1 || iGen->status() == 2 ) ){
-        
+	    
             TLorentzVector gen( iGen->px() , iGen->py() , iGen->pz() , iGen->energy() );
             TLorentzVector photon( iPhoton->px() , iPhoton->py() , iPhoton->pz() , iPhoton->energy() );
         
             if( gen.DeltaR(photon) < 0.2 ){ /// I LEFT OFF HERE!!!!!!
-              if( abs(iGen->mother()->pdgId()) > 100 && abs(iGen->mother()->pdgId()) != 2212 ) matchedGenNonPrompt++ ;
-              if( abs(iGen->mother()->pdgId()) <= 22 || abs(iGen->mother()->pdgId()) == 2212 ) matchedGenPrompt++ ;
-            }
+              if( abs(iGen->mother()->pdgId()) > 100 && abs(iGen->mother()->pdgId()) < 1000000 && abs(iGen->mother()->pdgId()) != 2212 ) matchedGenNonPrompt++ ;
+              if( abs(iGen->mother()->pdgId()) <= 100 || abs(iGen->mother()->pdgId()) == 2212 ){
+		if( gen.Pt()/photon.Pt() > 0.5 && gen.Pt()/photon.Pt() < 1.5 )
+		  matchedGenPrompt++ ;
+	      }//for prmopt photons
+            }//gen matching
           }
+	  //check wheather photon has matched to a gen electron or not
+	  if( abs(iGen->pdgId()) == 11 && iGen->status() == 1 ){
+	    TLorentzVector genE( iGen->px() , iGen->py() , iGen->pz() , iGen->energy() );
+            TLorentzVector photon( iPhoton->px() , iPhoton->py() , iPhoton->pz() , iPhoton->energy() );
+	    if( genE.DeltaR(photon) < 0.2 ){
```

Comment:
Actually never mind, I noticed this is done in other places that you didn't introduce, so I'll just fix it later